logging: Added support for 15 arguments in log message

nordic-krch opened the pull request:
Extended supported number of arguments in log message. Support for messages
consisting of more than 2 chunks had to be added. So far messages could
consist of one chunk (up to 3 args) or two chunks (2 args in first chunk and
7 in second chunk). Once 2+ chunks support is added number of arguments is
techinically limited to 15 (4 bit field). log_core and log_output extended
to suppor 10 arguments.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

jukkar reviewed (approved):
LGTM. Could we make this support max number of arguments already now, so that there would be no need to tweak the argument count later? At least in networking there are couple of places where we have more than 10 arguments, I made an workaround few patches ago but it wold be nice to get related prints to one output line.

nordic-krch committed:
logging: Support for up to 15 arguments in log message

Extended support in the log_core and log_output to 15 arguments
which is the hard limitation of log message format.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

spoorthik reviewed (commented):
Shippable is failing due to following error:

```
/opt/sdk/zephyr-sdk-0.9.3/sysroots/x86_64-pokysdk-linux/usr/libexec/i586-zephyr-elfiamcu/gcc/i586-zephyr-elfiamcu/6.2.0/real-ld: region `ROM' overflowed by 280 bytes
collect2: error: ld returned 1 exit status

```

Please have a look.

pizi-nordic commented on subsys/logging/log_output.c:
```diff
@@ -203,94 +203,83 @@ static void std_print(struct log_msg *msg,
 {
 	const char *str = log_msg_str_get(msg);
+	u32_t nargs = log_msg_nargs_get(msg);
+	u32_t args[nargs];
```
The VLA usually translates to malloc(). Please use alloca() here.

nordic-krch replied:
done
